v2/json2/server.go

```diff
@@ -186,22 +190,63 @@ func newCodecRequest(
 	}
 }
 
+// isBatch returns true when the first non-whitespace characters is '['
+func isBatch(raw json.RawMessage) bool {
+	for _, c := range raw {
+		// skip insignificant whitespace (http://www.ietf.org/rfc/rfc4627.txt)
+		if c == 0x20 || c == 0x09 || c == 0x0a || c == 0x0d {
+			continue
+		}
+		return c == '['
+	}
+	return false
```

Comment:
```suggestion
	return gjson.ParseBytes([]byte(raw)).IsArray()
```

Does exactly the same thing as you